FreeRTOS/FreeRTOS-Plus-TCP
# Send ARP packets for unknown IP addresses

Description
-----------
The IP stack should not populate its ARP cache with received datagrams and should only use ARP to create a connection between a MAC address and an IP-address.

Ref. RFC 1122 sec 2.3.3: "Address translation from Internet addresses to link-layer addresses on Ethernet and IEEE 802 networks MUST be managed by the Address Resolution Protocol (ARP)."

This is based on an earlier PR https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/318.

Test Steps
-----------
This can be tested by sending an ICMP ping to the FreeRTOS+TCP device from a device on local subnet. After this change, the unknown IP address will be queried through ARP first and then once resolved - the ICMP reply will be sent by the FreeRTOS+TCP stack.

Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
